lib/prism/translation/parser/lexer.rb

```diff
@@ -236,6 +239,13 @@ def to_a
             location = Range.new(source_buffer, offset_cache[token.location.start_offset], offset_cache[token.location.end_offset])
 
             case type
+            when :kDO
+              types = tokens.map(&:first)
+              nearest_lambda_token_type = types.reverse.find { |type| LAMBDA_TOKEN_TYPES.include?(type) }
```

Comment:
I'm not sure if this works for all cases.

There's some weirdness where you might have something like:

```ruby
-> (foo = -> (bar) {}) do end
```

would that impact this?